py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -153,18 +154,21 @@ def process_files(self) -> Iterable[pyarrow.table]:
                     process_logger.add_metadata(file_count=0)
                     process_logger.log_start()
 
+                # skip files that have been generated after the start of the
+                # current hour. don't add them to archive or error so that they
+                # are picked up next go around.
+                if timestamp >= self.start_of_hour:
+                    break
+
                 table = pyarrow.concat_tables([table, new_data])
                 self.archive_files.append(file)
                 file_count += 1
 
             except Exception:
                 self.error_files.append(file)
 
-        process_logger.add_metadata(
-            file_count=file_count, number_of_rows=table.num_rows
-        )
+        process_logger.add_metadata(file_count=0, number_of_rows=0)
         process_logger.log_complete()
-        yield table
```

Comment:
Remove this `yield` so we are only yielding tables on hour breaks, never in the middle of an hour